src/components/AddActionDialog.vue

```diff
@@ -1,11 +1,15 @@
 <script setup lang="ts">
+import type { ActionElement } from '@/action'
 import actionStore from '@/actionStore'
+import { cloneDeep } from '@/helpers'
 import { ref } from 'vue'
+import { v4 as guid } from 'uuid'
 
 const props = defineProps(['action'])
 const showAddNewDialog = ref(false)
-const actionName = ref(props.action?.Label ?? '')
 const actionId = ref(props.action?.Id ?? '')
+const actionName = ref(cloneDeep(props.action?.Label ?? ''))
+const actionElements = ref(cloneDeep(props.action?.Elements ?? []))
```

Comment:
`actionId` should also go through the clone process, or `Label` should not go through the cloning process, depends on how primitives are handled in vue